backend/src/news/router.py

```diff
@@ -119,4 +89,17 @@ async def search_news(request: PromptRequest):
             news_list.append(detailed_news)
         except Exception as e:
             print(e)
-    return sorted(news_list, key=lambda x: x["time"], reverse=True)
\ No newline at end of file
+    return sorted(news_list, key=lambda x: x["time"], reverse=True)
+
+@router.post("/news_summary_custom_model")
+async def get_news_summary_custom_model(
+    payload: NewsSumaryCustomModelSchema
+):
+    if(payload.model == "openai"):
+        llm_client = OPENAIClient()
+    elif(payload.model == "anthropic"):
+        llm_client = ANTHROPICClient()
+    else:
+        return "Invalid Client!"
+    response = llm_client.sum_up_news(payload.content)
+    return response
```

Comment:
The business logic of the endpoint `/news_summary_custom_model` has great similarities with `/news_summary`. I'd suggest wrapping them into a function which takes a `model` parameter to avoid repetition.